### PR TITLE
Add test coverage for amd64 MSBuild.exe

### PR DESCRIPTION
**Pull request by stan-sz**
MSBuild.exe ships with 64-bit binary. Test for it

### CODE REVIEW

File: src/E2ETests/E2ETests.cs [hunk 1]
```diff
@@ -10,12 +10,12 @@ public sealed class E2ETests
 {
     private readonly record struct Warning(string Message, string Project);
 
-    private static readonly (string ExePath, string Verb) MSBuild = GetMsBuildExeAndVerb();
-
     private static readonly Regex WarningErrorRegex = new(
         @".+: (warning|error) (?<message>.+) \[(?<project>.+)\]",
         RegexOptions.Compiled | RegexOptions.ExplicitCapture);
 
+    public static IEnumerable<object[]> MSBuildPlatform => new[] { new object[] { "" }, new object[] { "amd64" } };
```
Comment (stan-sz) [inline src/E2ETests/E2ETests.cs:17]: This is valid only for windows

**Comment by dfederm** [comment]
I'm not sure this is necessary, or at least not worth doubling the number of tests.

We probably should use the x64 version for testing since that's the default, but I don't think we need to explicitly test both of there's no platform-specific logic anywhere.

File: src/Tests/E2ETests.cs [hunk 2]
```diff
@@ -489,41 +490,6 @@ private static (string ExePath, string Verb) GetMsBuildExeAndVerb()
         return ("dotnet", "build");
     }
 
-    // From: https://docs.microsoft.com/en-us/dotnet/standard/io/how-to-copy-directories
-    private static void DirectoryCopy(string sourceDirName, string destDirName)
```
Comment (stan-sz) [inline src/Tests/E2ETests.cs:493]: Unused code

File: src/Tests/E2ETests.cs [hunk 3]
```diff
@@ -475,7 +475,8 @@ private static (string ExePath, string Verb) GetMsBuildExeAndVerb()
 
             if (!string.IsNullOrEmpty(vsInstallDir))
             {
-                string msbuildExePath = Path.Combine(vsInstallDir, @"MSBuild\Current\Bin\MSBuild.exe");
+                string msBuildPlatform = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "amd64" : string.Empty;
```
Comment (dfederm) [inline src/Tests/E2ETests.cs:478]: This is the msbuild.exe scenario, so this is always Windows.

**Review by dfederm — APPROVED** [review]

**Pull request merged**